src/codegen/codegen_neuron_cpp_visitor.cpp

```diff
@@ -1493,6 +1493,22 @@ void CodegenNeuronCppVisitor::print_nrn_init(bool skip_init_check) {
         printer->fmt_line("inst.{}[id] = v;", naming::VOLTAGE_UNUSED_VARIABLE);
     }
 
+    for (auto state: info.state_vars) {
+        auto state_name = state->get_name();
+        auto lhs_name = get_variable_name(state_name);
+        auto rhs_name = get_variable_name(state_name + "0");
+
+        if (!state->is_array()) {
+            auto state_name = state->get_name();
+            printer->fmt_line("{} = {};", lhs_name, rhs_name);
+        } else {
+            auto n_elements = state->get_length();
+            printer->fmt_push_block("for(size_t _i = 0; _i < {}; ++_i)", n_elements);
+            printer->fmt_line("{}[_i] = {};", lhs_name, rhs_name);
```

Comment:
Optional.  I wonder if it would be more clear to enclose the one line body of the for loop in ```{}```. (I realize I failed to do that with nocmodl code generation).

Reply:
When using `push_block` or `fmt_push_block` the `{` is implied. Similary `pop_block` will close the scope with a `}`. In the timeline for NMODL PRs there's often a thin item:

![2024-08-09-135545_1669x261_scrot](https://github.com/user-attachments/assets/d6a44622-12f4-49d2-a20b-028d77cc838a)

If you click on "References ..." it'll bring you to the `nmodl-references` repository. For every MOD file in `test/usecases` it will contain the generated `.cpp` files. Generally, once for NEURON and once for CoreNEURON. However, it's configurable, but we rare do so.

The generated code for NEURON is:
```
            inst.X[id] = inst.global->X0;
            inst.Y[id] = inst.global->Y0;
            inst.Z[id] = inst.global->Z0;
            for(size_t _i = 0; _i < 3; ++_i) {
                (inst.A+id*3)[_i] = inst.global->A0;
            }
            for(size_t _i = 0; _i < 2; ++_i) {
                (inst.B+id*2)[_i] = inst.global->B0;
            }
```